runtime/lua/vim/highlight.lua

```diff
@@ -0,0 +1,41 @@
+local api = vim.api
+
+local highlight = {}
+
+--- Highlight the yanked region
+--
+--- use from init.vim via
+---   au TextYankPost * lua require'vim.highlight'.on_yank()
+--- customize highlight group and timeout via
+---   au TextYankPost * lua require'vim.highlight'.on_yank("IncSearch", 500)
+-- @param higroup highlight group for yanked region
+-- @param timeout time in ms before highlight is cleared
+-- @param event event structure
+function highlight.on_yank(higroup, timeout, event)
+  event = event or vim.v.event
+  if event.operator ~= 'y' or event.regtype == '' then return end
+  higroup = higroup or "IncSearch"
+  timeout = timeout or 500
+
+  local bufnr = api.nvim_get_current_buf()
+  local yank_ns = api.nvim_create_namespace('')
+  api.nvim_buf_clear_namespace(bufnr, yank_ns, 0, -1)
+
+  local pos1 = vim.fn.getpos("'[")
```

Comment:
Shouldn't this use ``"`["`` (and similar for the end mark) instead since not all yanks are linewise?

Reply:
Do you have an example where this would make an actual difference? The current implementation handles all variations of yanking I could think of correctly.

Reply:
You're right.  I was mistaking accessing the values with jumping to the mark.  That is, `'[` and `` `[`` jump to different positions but `getpos("'[")` returns the correct data.